src/js/LineChart.jsx

```diff
@@ -0,0 +1,294 @@
+      .join('');
+
+    const x = this.getX();
+    const y = this.getY();
+    pathD +=
+      'L ' +
+      this.getSvgX(x.max) +
+      ' ' +
+      this.getSvgY(y.min) +
+      ' ' +
+      'L ' +
+      this.getSvgX(x.min) +
+      ' ' +
+      this.getSvgY(y.min) +
+      ' ';
+
+    return <path className="linechart_area" d={pathD} />;
+  }
+  // BUILD GRID AXIS
+  makeAxis() {
+    const { yLabelSize } = this.props;
+    const x = this.getX();
+    const y = this.getY();
+
+    return (
+      <g className="linechart_axis">
+        <line
+          x1={this.getSvgX(x.min) - yLabelSize}
+          y1={this.getSvgY(y.min)}
+          x2={this.getSvgX(x.max)}
+          y2={this.getSvgY(y.min)}
+          strokeDasharray="5"
+        />
+        <line
+          x1={this.getSvgX(x.min) - yLabelSize}
+          y1={this.getSvgY(y.max)}
+          x2={this.getSvgX(x.max)}
+          y2={this.getSvgY(y.max)}
+          strokeDasharray="5"
+        />
+      </g>
+    );
+  }
+  makeLabels() {
+    const {
+      svgHeight,
+      svgWidth,
+      xLabelSize,
+      yLabelSize,
+      duration,
+      coin
+    } = this.props;
+    const padding = 5;
+    return (
+      <g className="linechart_label">
+        {/* Y AXIS LABELS */}
+        <text
+          transform={`translate(${yLabelSize / 2}, 20)`}
+          textAnchor="middle"
+        >
+          {this.getY().max.toLocaleString('us-EN', {
+            style: 'currency',
+            currency: 'USD'
+          })}
+        </text>
+        <text
+          transform={`translate(${yLabelSize / 2}, ${svgHeight -
+            xLabelSize -
+            padding})`}
+          textAnchor="middle"
+        >
+          {this.getY().min.toLocaleString('us-EN', {
+            style: 'currency',
+            currency: 'USD'
+          })}
+        </text>
+        {/* X AXIS LABELS */}
+        <text
+          transform={`translate(${yLabelSize}, ${svgHeight})`}
+          textAnchor="start"
+        >
+          {this.props.data[coin][duration][0].d}
+        </text>
+        <text
+          transform={`translate(${svgWidth}, ${svgHeight})`}
+          textAnchor="end"
+        >
+          {[this.props.data[coin][duration].length - 1].d}
+        </text>
+      </g>
+    );
+  }
+  // FIND CLOSEST POINT TO MOUSE
+  getCoords(e) {
+    const { svgWidth, data, yLabelSize, duration, coin } = this.props;
+    const svgLocation = document
+      .getElementsByClassName('linechart')[0]
+      .getBoundingClientRect();
+    const adjustment = (svgLocation.width - svgWidth) / 2; //takes padding into consideration
+    const relativeLoc = e.clientX - svgLocation.left - adjustment;
+
+    let svgData = [];
+    data[coin][duration].map((point, i) => {
+      svgData.push({
+        svgX: this.getSvgX(point.x),
+        svgY: this.getSvgY(point.y),
+        d: point.d,
+        p: point.p
+      });
+    });
+
+    let closestPoint = {};
+    for (let i = 0, c = 500; i < svgData.length; i++) {
```

Comment:
Could you do this without a for loop?

Reply:
yeah, but it's quite neat like this!

Reply:
I'm not a fan of for loops but I'll let it slide 😉